src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -64,12 +63,12 @@ public static void Run(AddSourceArgs args, Func<ILogger> getLogger)
             var newPackageSource = new Configuration.PackageSource(args.Source, args.Name);
             newPackageSource.AllowInsecureConnections = args.AllowInsecureConnections;
 
-            if (newPackageSource.IsHttp && !newPackageSource.IsHttps && !newPackageSource.AllowInsecureConnections)
+            if (newPackageSource.IsHttp && !newPackageSource.IsHttps && !args.AllowInsecureConnections)
```

Comment:
I guess you have the option to check with the package source since that's what gets persisted. You're already assigning it anyways: args.AllowInsecureConnections. 

Not sure it matters too much. :D